### ISSUE TITLE
Make receipts command faster

### ISSUE THREAD

**Issue by cuducos** [datasciencebr/jarbas#77]
Multithreading and handling multiple requests in parallel are the key (it's the `Reimbursement.get_receipt_url()` from #76 that handles the HTTP requests)

**Comment by gwmoura** [comment]
@cuducos sorry, I am don't know much about this area, but searching by this subject I found this package: multiprocessing - https://docs.python.org/3/library/multiprocessing.html
> effectively side-stepping the Global Interpreter Lock by using subprocesses instead of threads.

Probabilly you know about this...

I found too this tutorial and I think he can help us - https://www.toptal.com/python/beginners-guide-to-concurrency-and-parallelism-in-python

**Comment by cuducos** [comment]
Yep, I used this on #66 — it works fine but I couldn't write tests for it (I mean, the tests I wrote were not working), so still on hold…

**Comment by cuducos** [comment]
Anyway… further thoughts: in the case of receipts, `asyncio` seems more promising than `multiprocessing` alone due to long time waiting for the response os a request (not long time waiting for processing). The trick is to start multiple HTTP requests ar once, not necessarily to distribute them among processes/threads ; )

**Comment by gwmoura** [comment]
So, we can start multiple HTTP requests using `multiprocessing` or threads and use `asyncio` to create an asynchronous HTTP consumer or `asyncio` solves the all problems?

**Comment by cuducos** [comment]
Actually I think that `asyncio` would be better to make parallel requests: a coroutine that starts the next request while waits for the current one's response, and so on… this way we can have a lot of requests started in the latency in-between sending a request and getting its response. Later we can start different process like that, but we have to test and see the results first IMHO…
